### ISSUE TITLE
isPasswordValidForArchive return true with wrong password

### ISSUE THREAD

**Issue by qchenqizhi** [ZipArchive/ZipArchive#633]
### Steps to reproduce
Using the zip package I created with command in macOS:
`zip -er test3.zip test.txt`
then set the password as "aaabbb".

try calling the `isPasswordValidForArchive` method for the zip file in iOS Simulator with Swift code:

```
        print(SSZipArchive.isPasswordValidForArchive(atPath: path, password: "1989", error: &error))
        // print true
        print(SSZipArchive.isPasswordValidForArchive(atPath: path, password: "aaabbb", error: &error))
        // print true
```

### Expected behavior
return true with wrong password

### Actual behavior
return false with wrong password
return true with right password

### Version of ZipArchive
2.4.2

### Environmental information

[test3.zip](https://github.com/ZipArchive/ZipArchive/files/8039339/test3.zip)

**Comment by Coeur** [comment]
confirmed...

Some notes here:
> uncompressed_size = 4
>  aes_version = 0

So it's only 4 bytes of data (which content is "test"), and not using AES.

I wonder why "1989" is accepted.

**Comment by Coeur** [comment]
The "guilty" code could be:
https://github.com/ZipArchive/ZipArchive/blob/be6ac340cbe394cc6a81eae6034a91a7702d9db3/SSZipArchive/minizip/mz_strm_pkcrypt.c#L166

Because it will skip the check when verify2 is 0, which happens to be the case with "1989" on that specific content "test".

**Comment by Coeur** [comment]
That line of code still exists in latest minizip:
https://github.com/zlib-ng/minizip-ng/blob/fe5fedc365f7824ada0cf9a84fb79b30d5fc97a8/mz_strm_pkcrypt.c#L166

And it was like that since year 2017:
https://github.com/zlib-ng/minizip-ng/commit/18a30653b1eb5e9f96ebc5ec1d93313d229bc731

You'll want to post your issue to https://github.com/zlib-ng/minizip-ng/issues and ping @nmoinvaz.

**Comment by Coeur** [comment]
I also found this on internet:
https://stackoverflow.com/questions/53552562/why-the-zip-appnote-on-pkware-website-didnt-mentioned-the-correct-password-chec

There seems to be an undocumented way for checking passwords.

**Comment by Coeur** [comment]
Nathan just fixed the bug, so I'll soon update minizip to get the fix in ZipArchive.

**Comment by Coeur** [comment]
Fixed in #724
